test/lib/resources/registry_test.rb

```diff
@@ -17,6 +17,8 @@ def test_pulp3_registry_url
       end
 
       def test_crane_registry_url
+        pulp_primary = FactoryBot.create(:smart_proxy, :default_smart_proxy, :with_pulp3)
+        ::SmartProxy.expects(:pulp_primary).at_least_once.returns(pulp_primary)
```

Comment:
@jlsherrill @ianballou this test is the one giving me a headache, and then I noticed that it stems from RegistryResource still supporting crane. Would I be better off first opening a PR to rip that chunk of code out? If, as I understand it, RegistryResource is for just talking to the registry on the primary Pulp server, we should never have crane present ever again at this point and this code is dead code.

Reply:
yes, +1 from me, we can rip this out

Reply:
Proposing doing clean up over here first -- https://github.com/Katello/katello/pull/9450